src/me/containers/MePage.tsx

```diff
@@ -21,19 +23,74 @@ import AlertsActivity from 'src/me/components/AlertsActivity'
 
 // Utils
 import {pageTitleSuffixer} from 'src/shared/utils/pageTitles'
-
+import {getOrg} from 'src/organizations/selectors'
+import {notify} from 'src/shared/actions/notifications'
+import {shouldShowUpgradeButton} from 'src/me/selectors'
 // Types
 import {AppState} from 'src/types'
 
 // Decorators
 import {ErrorHandling} from 'src/shared/decorators/errors'
 import {isFlagEnabled} from 'src/shared/utils/featureFlag'
 
+import {getUserWriteLimitHits} from 'src/cloud/components/onboarding/useGetUserStatus'
+import {writeLimitReached} from 'src/shared/copy/notifications'
+
+// Components
+import {UpgradeContent} from 'src/cloud/components/RateLimitAlertContent'
+import {dismissOverlay, showOverlay} from 'src/overlays/actions/overlays'
+
+const QUERY_WRITE_LIMIT_HITS = 100
+
 type ReduxProps = ConnectedProps<typeof connector>
 type Props = ReduxProps
 
+let hasCalled = false // We only want to show the write limit notification once
+
 @ErrorHandling
 export class MePage extends PureComponent<Props> {
+  async componentDidMount() {
```

Comment:
1. Should this be flagged?
2. re: the comment about this only wanting to be called once - what if the user hits their limit after this mounts? Also, why are we doing the same thing here as we are in the WriteLimits component?

Reply:
See comment below, this is a different implementation with a different use case